### ISSUE TITLE
BUG:  Cannot install Pandas 1.35 on MacOS Monterey with Chip M1

### ISSUE THREAD

**Issue by edmondo1984** [pandas-dev/pandas#44959]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
When installing panda 1.3.5 on MacOs 12.1 the compilation fails 


In file included from pandas/_libs/window/aggregations.cpp:747:
    In file included from ./pandas/_libs/src/headers/cmath:41:
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:321:9: error: no member named 'signbit' in the global namespace; did you mean 'sigwait'?
    using ::signbit;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/signal.h:101:5: note: 'sigwait' declared here
    int     sigwait(const sigset_t * __restrict, int * __restrict) __DARWIN_ALIAS_C(sigwait);
            ^
    In file included from pandas/_libs/window/aggregations.cpp:747:
    In file included from ./pandas/_libs/src/headers/cmath:41:
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:322:9: error: no member named 'fpclassify' in the global namespace
    using ::fpclassify;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:323:9: error: no member named 'isfinite' in the global namespace
    using ::isfinite;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:324:9: error: no member named 'isinf' in the global namespace
    using ::isinf;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:325:9: error: no member named 'isnan' in the global namespace
    using ::isnan;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:326:9: error: no member named 'isnormal' in the global namespace
    using ::isnormal;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:327:7: error: no member named 'isgreater' in the global namespace; did you mean '::std::greater'?
    using ::isgreater;
          ^~
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/functional:742:29: note: '::std::greater' declared here
    struct _LIBCPP_TEMPLATE_VIS greater : binary_function<_Tp, _Tp, bool>
                                ^
    In file included from pandas/_libs/window/aggregations.cpp:747:
    In file included from ./pandas/_libs/src/headers/cmath:41:
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:328:7: error: no member named 'isgreaterequal' in the global namespace; did you mean '::std::greater_equal'?
    using ::isgreaterequal;
          ^~
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/functional:771:29: note: '::std::greater_equal' declared here
    struct _LIBCPP_TEMPLATE_VIS greater_equal : binary_function<_Tp, _Tp, bool>
                                ^
    In file included from pandas/_libs/window/aggregations.cpp:747:
    In file included from ./pandas/_libs/src/headers/cmath:41:
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:329:9: error: no member named 'isless' in the global namespace
    using ::isless;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:330:9: error: no member named 'islessequal' in the global namespace
    using ::islessequal;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:331:9: error: no member named 'islessgreater' in the global namespace
    using ::islessgreater;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:332:9: error: no member named 'isunordered' in the global namespace
    using ::isunordered;
          ~~^
    /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/cmath:333:9: error: no member named 'isunordered' in the global namespace
    using ::isunordered;
          ~~^
    pandas/_libs/window/aggregations.cpp:4444:23: error: expected unqualified-id
        __pyx_t_1 = (std::signbit(__pyx_v_val) != 0);
                          ^
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/math.h:170:5: note: expanded from macro 'signbit'
        ( sizeof(x) == sizeof(float)  ? __inline_signbitf((float)(x))        \
        ^
    pandas/_libs/window/aggregations.cpp:4560:23: error: expected unqualified-id
        __pyx_t_1 = (std::signbit(__pyx_v_val) != 0);
                          ^
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/math.h:170:5: note: expanded from macro 'signbit'
        ( sizeof(x) == sizeof(float)  ? __inline_signbitf((float)(x))        \
        ^
    pandas/_libs/window/aggregations.cpp:5312:21: error: expected unqualified-id
      __pyx_t_1 = (std::isnan(__pyx_v_val) != 0);
                        ^
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/math.h:165:5: note: expanded from macro 'isnan'
        ( sizeof(x) == sizeof(float)  ? __inline_isnanf((float)(x))          \
        ^
    pandas/_libs/window/aggregations.cpp:15597:21: error: expected unqualified-id
      __pyx_t_1 = (std::isnan(__pyx_v_val) != 0);
                        ^
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/math.h:165:5: note: expanded from macro 'isnan'
        ( sizeof(x) == sizeof(float)  ? __inline_isnanf((float)(x))          \
        ^
    219 warnings and 17 errors generated.
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
    ERROR: Failed building wheel for pandas
  Failed to build pandas
  ERROR: Could not build wheels for pandas which use PEP 517 and cannot be installed directly
```
```


### Issue Description

Just run poetry add pandas on M1 - Monterey MacOs

### Expected Behavior

Installation would work :)

### Installed Versions

Trying to install 1.3.5

**Comment by JzHuang4** [comment]
Hi, do you have a problem to import pandas? It reports "miniforge3/envs/tf27/lib/python3.9/site-packages/pandas/_libs/interval.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e'))"

**Comment by gregnavis** [comment]
I also run into the same problem when trying to install pandas on Apple M1 with macOS Monterey 12.1 and Python 3.9.9 installed via MacPorts.

**Comment by MisLink** [comment]
It seems to be fixed in 1.4.0rc0.
